lib/lookup/src/lookup_buf/segmentbuf.rs

```diff
@@ -0,0 +1,145 @@
+use crate::*;
+use std::fmt::{Display, Formatter};
+use tracing::instrument;
+
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Clone, Hash)]
+pub struct FieldBuf {
+    pub name: String,
+    // This is a very lazy optimization to avoid having to scan for escapes.
+    pub requires_quoting: bool,
+}
+
+impl FieldBuf {
+    pub fn as_str(&self) -> &str {
+        &self.name
+    }
+}
+
+impl Display for FieldBuf {
+    fn fmt(&self, formatter: &mut Formatter<'_>) -> Result<(), std::fmt::Error> {
+        if self.requires_quoting {
+            write!(formatter, r#""{}""#, self.name)
+        } else {
+            write!(formatter, "{}", self.name)
+        }
+    }
+}
+
+impl From<String> for FieldBuf {
+    fn from(mut name: String) -> Self {
+        let requires_quoting = name.starts_with('\"');
+        if requires_quoting {
+            // There is unfortunately not way to make an owned substring of a string.
+            // So we have to take a slice and clone it.
+            let len = name.len();
+            name = name[1..len - 1].to_string();
+        }
+        Self {
+            name,
+            requires_quoting,
+        }
+    }
+}
```

Comment:
This too assumes a valid field format for the string. This should be `TryFrom` to guard against invalid formats.

Or, at the very least, the `requires_quoting` needs to be changed, so that any invalid non-quoted field gets turned into a quoted field.